packages/relay/src/routes/vote.ts

```diff
@@ -0,0 +1,193 @@
+import { DB } from 'anondb/node'
+import { Express } from 'express'
+import { EpochKeyProof } from '@unirep/circuits'
+import { errorHandler } from '../middleware'
+import { UnirepSocialSynchronizer } from '../synchornizer'
+import { VoteAction } from '../types'
+
+export default (
+    app: Express,
+    db: DB,
+    synchronizer: UnirepSocialSynchronizer
+) => {
+    app.post(
+        '/api/vote',
+        errorHandler(async (req, res, next) => {
+            await Vote(req, res, db, synchronizer)
+        })
+    )
+
+    // TODO: add get api to fetch all votes of post Id for frontend to check if user voted or not
+}
+
+/**
+ * Use this function to help us find out which vote action the user
+ * is going to do
+ *
+ * @param vote true for upvote action, false for downvote action
```

Comment:
I think it's not intuitive and not scalable to set vote as bool here (consider if we want more type of vote in the future). May use int or add another enum is better.

Reply:
I agree with you, I change the type of vote from `boolean` to `VoteAction`